### PR TITLE
Aarchiba kuiper

### PR DESCRIPTION
**Pull request by crawfordsm**
This is an updated version of #3724 by @aarchiba to allow merging of that pull request after minor fixes to merge conflicts and fix pep8.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -64,6 +64,8 @@ astropy.stats
 - Added false alarm probability computation to ``astropy.stats.LombScargle``
   [#6488]
 
+- Implemeteed Kuiper functions in ``astropy.stats`` [#3724]
```
Comment (pllim) [inline CHANGES.rst]: Typo, and wrong PR number listed, causing bot to complain.

File: astropy/stats/funcs.py [hunk 2]
```diff
@@ -1247,3 +1252,344 @@ def _kraft_burrows_nousek(N, B, CL):
         return _mpmath_kraft_burrows_nousek(N, B, CL)
 
     raise ImportError('Either scipy or mpmath are required.')
+
+
+def kuiper_FPP(D, N):
```
Comment (pllim) [inline astropy/stats/funcs.py]: It is generally frown upon to use uppercase for variable name (uppercase usually used for constant, not to be confused with `astropy.constants`). Is there a good reason to keep uppercase here?

**Comment by pllim** [comment]
I see that @hamogu did some review of the original PR, so I added him here as well.

File: CHANGES.rst [hunk 3]
```diff
@@ -64,6 +64,8 @@ astropy.stats
 - Added false alarm probability computation to ``astropy.stats.LombScargle``
   [#6488]
 
+- Implemented Kuiper functions in ``astropy.stats`` [#3724, #6565]]
```
Comment (pllim) [inline CHANGES.rst]: Please remove extra `]` 😅

**Comment by crawfordsm** [comment]
As this was previously reviewed and approved, and I've just updated this further, it would be good to have this merged sooner rather than later.   It looks like it has been passing all the relevant travis tests now.

**Comment by bsipocz** [comment]
Since this PR was an overtake by the submodule maintainer, and was reviewed before, thus I've approved it above to go ahead with the merging.

**Comment by mhvk** [comment]
@crawfordsm - the test here is one of the slowest of our suite. Does it really make sense to repeatedly test that the statistics produced by the routine are correct? Instead of running a routine 30000 times, I would think it makes more sense to check that for one or two realizations you get exactly the right answer (or, if that is not available, that the answer doesn't change from the routine that did have the right statistics, which we have now tested...). Given the same answer, the statistics cannot change, after all.

**Comment by bsipocz** [comment]
@mhvk - Can we open a separate issue for this? I vaguely remembered your comment, but took me a long time to find it again. 
I hope that the test can be optimized as currently it's indeed a unnecessary burden on the CI.

**Comment by mhvk** [comment]
@bsipocz - see #6980

**Pull request merged**